index.js

```diff
@@ -6,20 +6,55 @@
 const functionFlexIntervals = (interface) => {
 
   var config;
+  // show the image for at least two seconds and
+  // calculate the real value as soon as the config is available
+  var minChangeTime = 2000;
   var counter = 1;
   var isPaused = false;
   var changeTimeout = 0;
+  var nextChangeTime = 0;
+
+  /**
+   * Triggers the next image change if not paused
+   * @param {number} additionalChange millisecond
+   */
+  const setChangeTimer = (additionalChange) => {
+    clearTimeout(changeTimeout);
+    if (!isPaused ) {
+      // to enable logging execute: `~/TeleFrame/tools/addon_control flexInterval config logNextChangeInterval true`
+      if (interface.config.logNextChangeInterval) {
+        interface.logger.info("additional change in: " + additionalChange);
+      }
+      changeTimeout = setTimeout(() => {
+        // show next image only if there was no user interaction
+        if (!isPaused && new Date().getTime() >= nextChangeTime) {
+          // use the half of the configured fadeTime
+          interface.sendEvent('next', config.fadeTime / 2);
+        }
+      }, additionalChange);
+    }
+  };
+
 
   interface.registerListener('teleFrame-ready', teleFrameObjects => {
     config = teleFrameObjects.config;
+    minChangeTime = Math.max(2000, Math.round(config.interval/config.imageCount));
   });
   //interface.registerListener('newImage', (sender, type) => counter = 1);
 
   interface.registerListener(['images-loaded', 'imageDeleted', 'paused', 'muted',
     'recordStopped', 'unstarImage', 'starImage'],
-    () => counter = 1);
+    () => {
+      counter = 1;
+      // trigger next change
+      const now = new Date().getTime();
+      if ((nextChangeTime - now) > minChangeTime) {
```

Comment:
why do we have this if-clause? shouldnt it always start with minChangeTime (so it is not too short neither)?